torch/_inductor/fx_passes/pre_grad.py

```diff
@@ -92,6 +96,25 @@ def pre_grad_passes(gm: torch.fx.GraphModule, example_inputs):
     gm.graph.lint()
     gm.recompile()
 
+    # need to topo-sort graphmodule before we run the model,
+    # otherwise it may fail as refer before def
+    if config.pattern_matcher and config.runtime_numeric_check["enable"]:
```

Comment:
Let's refactor L99 to L117 into a function called ```numeric_check_if_enabled(gm_before_fx_passes, gm_after_fx_passes)``` and move the definition into ```numeric_utils.py```. I think that would make this file more clear logically.